apps/meteor/tests/end-to-end/api/incoming-integrations.ts

```diff
@@ -680,4 +683,267 @@ describe('[Incoming Integrations]', () => {
+					integration2 = res.body.integration;
+				});
+
+			await request
+				.post(api('integrations.create'))
+				.set(credentials)
+				.send({
+					type: 'webhook-incoming',
+					name: 'Incoming test 3 - Sending Messages',
+					enabled: true,
+					alias: 'Incoming test 3 - Sending Messages',
+					username: nonMemberUser.username as string,
+					scriptEnabled: false,
+					overrideDestinationChannelEnabled: false,
+					channel: `#${publicChannelInPrivateTeam.fname}`,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('integration').and.to.be.an('object');
+					integration3 = res.body.integration;
+				});
+			await request
+				.post(api('integrations.create'))
+				.set(credentials)
+				.send({
+					type: 'webhook-incoming',
+					name: 'Incoming test 4 - Sending Messages',
+					enabled: true,
+					alias: 'Incoming test 4 - Sending Messages',
+					username: nonMemberUser.username as string,
+					scriptEnabled: false,
+					overrideDestinationChannelEnabled: false,
+					channel: `#${publicRoom.fname}`,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('integration').and.to.be.an('object');
+					integration4 = res.body.integration;
+				});
+		});
+
+		after(async () => {
+			await Promise.all(
+				[publicRoom, privateRoom, publicChannelInPrivateTeam].map((room) => deleteRoom({ type: room.t as 'c' | 'p', roomId: room._id })),
+			);
+			await deleteTeam(credentials, privateTeam.name);
+			await deleteUser(nonMemberUser);
+			await Promise.all([
+				...[integration2, integration3, integration4].map((integration) =>
+					request.post(api('integrations.remove')).set(credentials).send({
+						integrationId: integration._id,
+						type: integration.type,
+					}),
+				),
+				updatePermission('manage-incoming-integrations', ['admin']),
+			]);
+		});
+
+		it('should not send a message to a private rooms on behalf of a non member', async () => {
+			const successfulMesssage = `Message sent successfully at #${Random.id()}`;
+			await request
+				.post(`/hooks/${integration2._id}/${integration2.token}`)
+				.send({
+					text: successfulMesssage,
+				})
+				.expect(400)
+				.expect((res) => {
+					expect(res.body).to.have.property('error', 'error-not-allowed');
+				});
+			await request
+				.get(api('groups.messages'))
+				.set(credentials)
+				.query({
+					roomId: privateRoom._id,
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('messages').and.to.be.an('array');
+					expect(!!(res.body.messages as IMessage[]).find((m) => m.msg === successfulMesssage)).not.to.be.true;
```

Comment:
Any reason not to do it like this?

```suggestion
					expect((res.body.messages as IMessage[]).find((m) => m.msg === successfulMesssage)).to.be.undefined;
```

or even something checking if the response list includes the successfulMessage;

Same thing goes for all the other checks that used a `!!`